hooks/context_change.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Context change hook.
+"""
+
+from tank import get_hook_baseclass
+
+
+class ContextChangeHook(get_hook_baseclass()):
+    """
+    Hook that gets executed every time there is a context change, once before
+    the change and once after.
+    """
+
+    def pre_context_change(self, current_context, next_context):
+        """
+        Called before the context change happens.
```

Comment:
The way the code is at the moment it's too late. It may be that the previous engine has already been torn down and there is no going back. We could redesign all of this of course.